packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp

```diff
@@ -0,0 +1,131 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Illustrates various property types for models")
+@scenarioService("/type/property/additionalProperties")
+namespace Type.Property.AdditionalProperties;
+
+@doc("Template to have models operations")
+interface ModelOperations<TModel, TDoc extends string> {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing"
+  @scenario
+  @scenarioDoc(
+    """
+  Expected response body:
+  ```json
+  {doc}
+  ```
+  """,
+    {
+      doc: TDoc,
+    }
+  )
+  @get
+  @doc("Get call")
+  get(): TModel;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing"
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+  @scenario
+  @scenarioDoc(
+    """
+  Expected input body:
+  ```json
+  {doc}
+  ```
+  """,
+    {
+      doc: TDoc,
+    }
+  )
+  @put
+  @doc("Put operation")
+  put(@body @doc("body") body: TModel): void;
+}
+
+// ********************************************** Record<unknown> **********************************************
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+@doc("The model extends from Record<unknown> type.")
+model ExtendsUnknownAddtionalProperties extends Record<unknown> {
+  @doc("The name property")
+  name: string;
+}
+
+@route("/extendsRecordUnknown")
+@operationGroup
```

Comment:
Shall we group them with "extends or is" instead of group them one by one?

Reply:
each interface has two operation: put and get. I follow the same pattern like value-types.